### PR TITLE
hierarchical Sorting

### PR DESCRIPTION
**Pull request by bikramkawan**
@mijar @katush Could you try to see on your side.

### CODE REVIEW

**Comment by bikramkawan** [comment]
I have made some changes on my phovea core. If the above PR does not work please comment this line and remove the comment in above block.
https://github.com/Caleydo/mothertable/blob/feature_103/src/column/ColumnManager.ts#L147
let me know if there is error in your pc.

**Comment by bikramkawan** [comment]
Address #103

**Comment by mstreit** [comment]
@bikramkawan, please fix the conflict. See failed checks.

**Comment by bikramkawan** [comment]
I have reverted to the concat method for joining the array of the ranges.

**Comment by mstreit** [comment]
please also report WHY you had to revert to an old state.

**Comment by bikramkawan** [comment]
I am waiting for the PR request.
https://github.com/phovea/phovea_core/pull/84

**Comment by katush** [comment]
@bikramkawan The hierarchical sorting doesn't work. The sorting only works for single column. If I have multiple columns and try to sort, the sorting removes some items from columns. Any idea what's happening?

**Comment by bikramkawan** [comment]
I think that is the issue related to #119. Could you try for the attributes which have all values for now. ?

**Comment by katush** [comment]
I tried - and it's still an issue. The missing values are removed even when I only have one column, but when I have multiple columns, the data are removed independent of that.

**Comment by bikramkawan** [comment]
Could you step down your use case here and screen dump?

**Comment by katush** [comment]
I added "in business" and "origin" and then tried to sort by "in business". It resulted in only one item remaining. @mijar can you try to run this as well? Bikram showed me that it works for him.
image

**Comment by mijar** [comment]
It is working for me....maybe @katush you have to update everything and start with everything in the new state...I see it like this:
hierarchicalok

File: src/column/ColumnManager.ts [hunk 1]
```diff
@@ -122,12 +129,37 @@ export default class ColumnManager extends EventHandler {
...
-    const indexes = (<any>data).indices;
-    const currentData = await (<any>data).idView(indexes.intersect(this.rangeNow));
-    const s = new SortColumn(currentData, sortMethod);
-    const r:Range1D = await s.sortMe();
-    this.update(r);
+  async onSortMethod(evt: any, sortMethod: string) {
+
+    this.sortMethod = sortMethod;
+    const cols: any = this.columns.filter((d) => d.data.desc.type === 'vector');
+    const s = new SortColumn(cols, this.sortMethod);
```
Comment (katush) [inline src/column/ColumnManager.ts]: Is it necessary to instantiate every time?  The object is not used afterwords.

File: src/sortColumn/SortColumn.ts [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+  }
+
+  async sortString(data) {
+    const sortedView = await data.sort(stringSort.bind(this, this.sortCriteria));
+    const sortedRange = await  sortedView.ids();
+    return sortedRange;
+
+  }
+
+  async getCatRange(col) {
```
Comment (katush) [inline src/sortColumn/SortColumn.ts]: This and getNumericalRange are basically same - refactor please.

**Pull request merged**